### ISSUE TITLE
GELF log level seems not working

### ISSUE THREAD

**Issue by hillout** [keycloak/keycloak#24358]
### Before reporting an issue

- [X] I have searched existing issues
- [X] I have reproduced the issue with the [latest nightly release](https://github.com/keycloak/keycloak/releases/tag/nightly)

### Area

admin/cli

### Describe the bug

Hi, gelf level logging seems not working. 

KC in a docker container.

A part of .env file:
```
KC_LOG_GELF_HOST=tc-logstash0.domain.com
KC_LOG_GELF_PORT=12207
KC_LOG_GELF_LEVEL=DEBUG
KC_LOG_GELF_FACILITY=tc-kc02-cluster0
```

Using these environments I still cal see only **INFO** logs and nothing more.  

I tried another approach - docker compose: 
```start --log="console,gelf" --log-level="INFO,org.keycloak.events:debug" --log-gelf-level="debug" --log-gelf-host=tc-msk-logstash0.domain.com --log-gelf-port=12207 --log-gelf-facility=tc-kc02-cluster0```

No luck as well, only **INFO** logs I can see in ELK. 

Also I tried  ```--log-gelf-level="INFO,org.keycloak.events:debug"```  - but with this key Keycloak **didn't start at all.** 

**BTW**,  why there is no env for simple KC_LOG_LEVEL ? Only for KC_LOG_GELF_LEVEL - which is not working as I described above (https://www.keycloak.org/server/all-config?q=gelf). 

I ended up using standard docker logging option:

```    
  logging:
      driver: gelf
      options:
        gelf-address: udp://tc-msk-logstash0.domain.com:12207
``` 

### Version

22.0.5

### Expected behavior

I would see DEBUG logs in ELK.

### Actual behavior

I see only INFO logs.

### How to Reproduce?

Follow KC instruction at https://www.keycloak.org/server/logging#_example_setup_using_the_elk_stack page. Use GELF log level keys like I am described above. As I noticed DEBUG logs don't appear if I use **KC_LOG_GELF_FACILITY** parameter.

**Comment by shawkins** [comment]
> BTW, why there is no env for simple KC_LOG_LEVEL ? Only for KC_LOG_GELF_LEVEL - which is not working as I described above (https://www.keycloak.org/server/all-config?q=gelf).

There is: https://www.keycloak.org/server/all-config?q=log-level 

That should be what is missing in this scenario for you to see the debug logs in gelf.  Use KC_LOG_LEVEL=debug, or whatever list of individual context setting you need in addition to setting the KC_LOG_GELF_LEVEL.

**Comment by hillout** [comment]
@shawkins hi! Well, at some point in the past when I was written the subject there wasn't info about KC_LOG_LEVEL on the official site (I was confused, because I remember it certainly was there before). Anyway, once I discovered it was back, I used this flag and now my .env file is:

```bash
KC_LOG=console,gelf
KC_LOG_LEVEL="INFO,org.keycloak.events:debug"
KC_LOG_GELF_HOST=logstash0.domain.com
KC_LOG_GELF_PORT=12207
KC_LOG_GELF_LEVEL="DEBUG"
```

So, I see in the stdout logs for org.keycloak.events:debug, but I don't see them in GELF while catching in ELK.  

I will again highlight the point that it is not possible to set such an option in GELF: "KC_LOG_GELF_LEVEL=INFO,org.keycloak.events:debug", KC doesn't start at all. 

Logs are saying:

```
The config property quarkus.log.handler.gelf.level with the config value "INFO,org.keycloak.events:debug" threw an Exception whilst being converted Unknown level "INFO,ORG.KEYCLOAK.EVENTS:DEBUG"
```

**Comment by shawkins** [comment]
> So, I see in the stdout logs for org.keycloak.events:debug, but I don't see them in GELF while catching in ELK.

But you do see the info and above logs?  Out of curiosity does anything change if you use KC_LOG_GELF_LEVEL="ALL" - that matches the quarkus default behavior.

> I will again highlight the point that it is not possible to set such an option in GELF

That is correct, the property is documented as a single value, not a list.  That value looks like it needs to be a level.  So it's not possible to set things on a per context basis.  Are you requesting that as an enhancement?

**Comment by hillout** [comment]
@shawkins hi,
At the beginning of the issue I noticed, that not all logs appear sending over GELF if you put `--log-gelf-facility=kc02-cluster0`, only INFO, even though I explicitly set KC_LOG_GELF_LEVEL=DEBUG. That was the main issue here. 

I believe there is no need for rising an enhancement for GELF, although would've been handy if it was implemented, giving more flexibility over picking which logs are really needed.  

So then, I got rid out of `--log-gelf-facility=kc02-cluster0` option, and now I see the full stack of DEBUG logs in ELK.
